saleor/extensions/manager.py

```diff
@@ -178,7 +185,101 @@ def postprocess_order_creation(self, order: "Order"):
+    ) -> "GatewayResponse":
+        method_name = "capture_payment"
+        return self.__run_payment_method(gateway, method_name, payment_information)
+
+    def refund_payment(
+        self, gateway: Gateway, payment_information: "PaymentData"
+    ) -> "GatewayResponse":
+        method_name = "refund_payment"
+        return self.__run_payment_method(gateway, method_name, payment_information)
+
+    def void_payment(
+        self, gateway: Gateway, payment_information: "PaymentData"
+    ) -> "GatewayResponse":
+        method_name = "void_payment"
+        return self.__run_payment_method(gateway, method_name, payment_information)
+
+    def confirm_payment(
+        self, gateway: Gateway, payment_information: "PaymentData"
+    ) -> "GatewayResponse":
+        method_name = "confirm_payment"
+        return self.__run_payment_method(gateway, method_name, payment_information)
+
+    def process_payment(
+        self, gateway: Gateway, payment_information: "PaymentData"
+    ) -> "GatewayResponse":
+        method_name = "process_payment"
+        return self.__run_payment_method(gateway, method_name, payment_information)
+
+    def create_payment_form(self, data, gateway, payment_information):
+        method_name = "create_form"
+        return self.__run_payment_method(
+            gateway, method_name, payment_information, data=data
+        )
+
+    def get_client_token(self, gateway, token_config: "TokenConfig") -> str:
+        method_name = "get_client_token"
+        default_value = None
+        gateway_name = gateway.value
+        gtw = self.get_plugin(gateway_name)
+        return self.__run_method_on_single_plugin(
+            gtw, method_name, default_value, token_config=token_config
+        )
+
+    def list_payment_sources(
+        self, gateway: Gateway, customer_id: str
+    ) -> List["CustomerSource"]:
+        default_value = []
+        gateway_name = gateway.value
+        gtw = self.get_plugin(gateway_name)
+        if gtw is not None:
+            return self.__run_method_on_single_plugin(
+                gtw, "list_payment_sources", default_value, customer_id=customer_id
+            )
+        raise Exception(f"Payment plugin {gateway_name} is inaccessible!")
+
+    def list_payment_gateways(self) -> List[Gateway]:
+        PAYMENT_METHOD = "process_payment"
+        return [
+            Gateway(plugin.PLUGIN_NAME)
+            for plugin in self.plugins
+            if PAYMENT_METHOD in type(plugin).__dict__
+            and self.get_plugin_configuration(plugin.PLUGIN_NAME).active
+        ]
+
+    def __run_payment_method(
+        self,
+        gateway: Gateway,
+        method_name: str,
+        payment_information: "PaymentData",
+        **kwargs,
+    ) -> Optional["GatewayResposne"]:
+        default_value = None
+        gateway_name = gateway.value
+        gtw = self.get_plugin(gateway_name)
+        if gtw is not None:
+            return self.__run_method_on_single_plugin(
```

Comment:
What about the scenario when someone creates PaymentPlugin but without mandatory method? We don't have any validation here. It will just do nothing. Shouldn't we have some validation here?

Reply:
Sure, attempt to run _authorize_ or other payment method that we expect side effects should not fail silently. Changed.